File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/NhaCungCapColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { NhaCungCapRow } from "./NhaCungCapRow";

export interface NhaCungCapColumns {
    MaNcc: Column<NhaCungCapRow>;
    TenCongTy: Column<NhaCungCapRow>;
    TenLienHe: Column<NhaCungCapRow>;
    ChucDanhLienHe: Column<NhaCungCapRow>;
    DiaChi: Column<NhaCungCapRow>;
    ThanhPho: Column<NhaCungCapRow>;
    MaBuuChinh: Column<NhaCungCapRow>;
    QuocGia: Column<NhaCungCapRow>;
    SoDienThoai: Column<NhaCungCapRow>;
    MaSoThue: Column<NhaCungCapRow>;
    TrangChu: Column<NhaCungCapRow>;
    CreatedAt: Column<NhaCungCapRow>;
}

export class NhaCungCapColumns extends ColumnsBase<NhaCungCapRow> {
    static readonly columnsKey = 'SanPhamDB.NhaCungCap';
    static readonly Fields = fieldsProxy<NhaCungCapColumns>();
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/NhaCungCap/NhaCungCapColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.NhaCungCap")]
[BasedOnRow(typeof(NhaCungCapRow), CheckNames = true)]
public class NhaCungCapColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MaNcc { get; set; }
    [EditLink]
    public string TenCongTy { get; set; }
    public string TenLienHe { get; set; }
    public string ChucDanhLienHe { get; set; }
    public string DiaChi { get; set; }
    public string ThanhPho { get; set; }
    public string MaBuuChinh { get; set; }
    public string QuocGia { get; set; }
    public string SoDienThoai { get; set; }
    public string MaSoThue { get; set; }
    public string TrangChu { get; set; }
    public DateTime CreatedAt { get; set; }
}